staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go

```diff
@@ -229,15 +291,43 @@ L:
 		}
 	}
 
+	return events
+}
+
+// collectLastEvents assumes that the buffer was closed. It collects the first
+// maxBatchSize events from the closed buffer into a batch and returns them.
+func (b *batchBackend) collectLastEvents() []auditinternal.Event {
```

Comment:
It simplifies what can and does happen to the object (i.e., the call is a functional-style). As a reviewer I can scan the function to see that it doesn't mutate the object in any way. All the things it needs can be satisfied from the functions parameters. Equally, if I come to modify this file I have a reduced set of functions to consider where state is held and mutated.